src/renderers/shaders/ShaderChunk/worldpos_vertex.glsl

```diff
@@ -1,4 +1,4 @@
-#if defined( USE_ENVMAP ) || defined( DISTANCE ) || defined ( USE_SHADOWMAP )
+#if defined( USE_ENVMAP ) || defined( DISTANCE ) || defined ( USE_SHADOWMAP ) || defined ( PHYSICAL )
```

Comment:
This is not the proper solution. We can't be modifying the library to accommodate an example.

I expect you need to modify `NodeMaterial.js`, instead.

Reply:
Agreed!

Reply:
Of course, sorry for that. However in that case, I don't really know what's best next.  
Either turn on shadows for the two `DirectionalLight` in there and gain `#define USE_SHADOWMAP`, but that results in randomly occuring shader errors (`error X3531: can't unroll loops marked with loop attribute`) and warnings and seems forced.  
Or somehow replace `vWPosition = worldPosition.xyz` with `vWPosition = ( modelMatrix * vec4( transformed, 1.0 ) ).xyz`, which is what `worldPosition` would equate to. Although that would be simply ignoring the question of the missing `#define` though.
Switching from [`WORLD` to `LOCAL`](https://github.com/mrdoob/three.js/blob/0e8c0865c03877fcb9d5ecf4f1c821acd2bdebde/examples/webgl_materials_nodes.html#L1408-L1411) works as well, or at least the caustics effect doesn't seem to mind.  
Unfortunately, anything more complicated than that and I'm afraid I'm out of my league 😕